### ISSUE TITLE
central policy exporter dashboard for multi cluster

### ISSUE THREAD

**Issue by rshiva777** [fjogeleit/policy-reporter#167]
I deployed a set of kyverno with policies, policy exporter and policy exporter UI on cluster A, able to see the policy reports from UI

![image](https://user-images.githubusercontent.com/42074614/179021069-eec619d5-371e-416a-8138-a796181c3e95.png)

Configured one more setup of  kyverno with policies, policy exporter on cluster B, but this time without policy expoter UI. In helm chart values.yaml of policy exporter, for UI url field, i gave fqdn of cluster A policy UI url.

After installing of the setup,  i see the reports are pushed. I able to see report error from policy exporter log, but unable to see from dashbaord by filtering cluster or namespace etc.. How can we do this multi cluster UI setup

**Comment by fjogeleit** [comment]
Hey @rshiva777, currently the Policy Reporter Dashboards have no multi cluster support. The URL you mentioned are only for pushes which are shown in the "Logs" page but not for the dashboards.

The UI was intended as simple and standalone  alternative to a complete monitoring stack like Prometheus/Grafana.

Perhaps this could be a future feature, but for now it is not possible.

**Comment by rshiva777** [comment]
Thanks @fjogeleit 

Hope this feature will be implemented in near future.

**Comment by rshiva777** [comment]
Would this feature be implemented in upcoming release?

**Comment by fjogeleit** [comment]
It is nothing that I had on the roadmap right now because it is no usual use case.

Some questions about your expectations:

to make this work you would expose the Policy Reporter API from Cluster B in some way that Cluster A can access it. How would you secure the API, so that it is not available for the outside world. Because this should not be happen.

I also would not mix up the results from different clusters, I would provide some way to select a configured cluster and see the results from this selected one. Would this okay for your use case?

**Comment by rshiva777** [comment]
Thanks @fjogeleit  for the inputs.

Please suggest the best way to view a central dashboard to view policy reports from multiple cluster.

**Comment by fjogeleit** [comment]
Currently the only way would be a central grafana / prometheus for the provided metrics.

Other ways of multi cluster support are not available.

What are your thoughts about my questions above?

**Comment by rshiva777** [comment]
Hi @fjogeleit ,

to make this work you would expose the Policy Reporter API from Cluster B in some way that Cluster A can access it. How would you secure the API, so that it is not available for the outside world. Because this should not be happen.  -  **we make whitelist sources of policy reporter API at ingress level so that only it can be accessible by cluster A**

Is it a good option to be used?

**Comment by fjogeleit** [comment]
Okay, just want to be sure that it is possible to expose the API in a safe way. I created a first draft, for configuring additional cluster. I try to get some feedback and release it in a week or so. You can have a look to the preview if this would work for you.

The configuration will look like this:

```yaml
clusters:
- name: K3S Cluster
  api: http://localhost:8081
  kyvernoApi: http://localhost:8085
```

https://user-images.githubusercontent.com/16627596/179411219-e6bbb8ba-242b-428c-bccf-6d61b6cf1335.mp4

**Comment by rshiva777** [comment]
Yes, this is the functionality i am expecting. To be able to view multi-cluster policies from a single place.

**Comment by fjogeleit** [comment]
@rshiva777 I released v2.10.2 of the Policy Reporter Helm Chart with the new UI feature for additional external Clusters.

You can checkout the Documentation about the configuration: https://kyverno.github.io/policy-reporter/guide/helm-chart-core#external-clusters

Let me now if you have any issues.

**Comment by rshiva777** [comment]
sure, Thanks @fjogeleit  for the quick support
